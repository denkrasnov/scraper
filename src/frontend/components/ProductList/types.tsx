import { Product } from "~app/services/fetchProducts/types";

export interface ProductListProps {
  products: Product[];
}
