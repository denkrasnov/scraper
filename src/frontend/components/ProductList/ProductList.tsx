import React, { FC } from "react";
import useInfiniteScroll from "infinite-scroll-react-hook";

import Box from "~app/atoms/Box";
import isDesktop from "~app/atoms/hooks/isDesktop";
import ProductCard from "../ProductCard";
import { ProductListProps } from "./types";

const ProductList: FC<ProductListProps> = (props) => {
  const { products = [] } = props;
  const [items, setRef] = useInfiniteScroll(products);
  const desktop = isDesktop();

  return (
    <Box
      display="block"
      gridArea="result"
      margin={desktop ? "s20" : undefined}
      marginTop={desktop ? undefined : "s8"}
    >
      {items.map((product) => {
        return (
          <Box key={product.id} marginBottom="s8">
            <ProductCard
              channel={"No channel yet"}
              date={product.postDate}
              header={product.title}
              imageURL={product.imageURL}
              // newsURL={product.newsURL}
            />
          </Box>
        );
      })}
      <Box ref={setRef} bottom="200px" position="relative" />
    </Box>
  );
};

export default ProductList;
