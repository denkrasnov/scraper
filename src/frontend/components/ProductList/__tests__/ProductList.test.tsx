import React from "react";
import { shallow } from "enzyme";
import useInfiniteScroll from "infinite-scroll-react-hook";

import ProductList from "..";

jest.mock("infinite-scroll-react-hook", () => jest.fn(() => [[], () => {}]));

describe("ProductCard", () => {
  const products = [
    {
      imageURL: "__IMAGE_URL__",
      date: "9:00",
      header: "__TITLE__",
      newsUrl: "__URL__"
    }
  ];
  (useInfiniteScroll as jest.Mock).mockReturnValue([products, () => {}]);

  it("should render", () => {
    const component = shallow(<ProductList products={products} />);

    expect(component).toMatchSnapshot();
  });
});
