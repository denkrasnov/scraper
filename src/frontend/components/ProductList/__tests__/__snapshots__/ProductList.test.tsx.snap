// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render 1`] = `
<Box
  display="block"
  gridArea="result"
  margin="s20"
>
  <Box
    key="__TITLE__"
    marginBottom="s8"
  >
    <ProductCard
      channel="TV8"
      date="9:00"
      header="__TITLE__"
      imageURL="__IMAGE_URL__"
    />
  </Box>
  <Box
    bottom="200px"
    position="relative"
  />
</Box>
`;

exports[`ProductCard should render when mobile 1`] = `
<Box
  display="block"
  gridArea="result"
  marginTop="s8"
>
  <Box
    key="__TITLE__"
    marginBottom="s8"
  >
    <ProductCard
      channel="TV8"
      date="9:00"
      header="__TITLE__"
      imageURL="__IMAGE_URL__"
    />
  </Box>
  <Box
    bottom="200px"
    position="relative"
  />
</Box>
`;

exports[`ProductCard should render when no products prop 1`] = `
<Box
  display="block"
  gridArea="result"
  margin="s20"
>
  <Box
    key="__TITLE__"
    marginBottom="s8"
  >
    <ProductCard
      channel="TV8"
      date="9:00"
      header="__TITLE__"
      imageURL="__IMAGE_URL__"
    />
  </Box>
  <Box
    bottom="200px"
    position="relative"
  />
</Box>
`;
